app/components/MusicPlayerBar.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { motion } from 'framer-motion';
+import type { YouTubeEvent, YouTubePlayer } from 'react-youtube';
+import type { SvgIconComponent } from '@mui/icons-material';
+import {
+  PlayArrow,
+  Pause,
+  SkipNext,
+  SkipPrevious,
+  VolumeUp,
+  VolumeMute,
+  VolumeDown,
+  Shuffle,
+  Repeat,
+  Link as LinkIcon,
+  RepeatOne,
+  VolumeOff,
+  ArrowDropUpRounded,
+} from '@mui/icons-material';
+import Marquee from 'react-fast-marquee';
+import { useMediaQuery } from 'react-responsive';
+import usePlayList from '~/hooks/usePlayList';
+import PlayerPage from '@components/PlayerPage';
+
+const MusicPlayerBar: React.FC = () => {
+  const isMobile = useMediaQuery({ query: '(max-width: 767px)' });
+
+  const playlist = usePlayList();
+  const videoInfo = useMemo(() => playlist.getCurrentMusic(), [playlist]);
+
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [player, setPlayer] = useState<YouTubePlayer>();
+  const [currentTime, setCurrentTime] = useState(0);
+  const [duration, setDuration] = useState(0);
+  const [volume, setVolume] = useState(100);
+  const [isMuted, setIsMuted] = useState(false);
+  const [repeatMode, setRepeatMode] = useState<'NO_REPEAT' | 'REPEAT_ONE' | 'REPEAT_ALL'>('NO_REPEAT');
+  const [isOpen, setOpen] = useState(false);
+  const [shuffle, setShuffle] = useState(false);
+
+  const [timer, setTimer] = useState<any>();
+
+  const percent = useMemo(() => (currentTime / duration) * 100, [currentTime, duration]);
+
+  const PlayOrPause: SvgIconComponent = useMemo(() => isPlaying ? Pause : PlayArrow, [isPlaying]);
+
+  const VolumeIcon: SvgIconComponent = useMemo(() => {
+    if (isMuted) return VolumeOff;
+    if (volume === 0) return VolumeMute;
+    else if (volume < 50) return VolumeDown;
+    else return VolumeUp;
+  }, [isMuted, volume]);
+
+  const RepeatIcon: SvgIconComponent = useMemo(() => repeatMode === 'REPEAT_ONE' ? RepeatOne : Repeat, [repeatMode]);
+
+  const onReady = useCallback((event: YouTubeEvent) => {
+    setPlayer(event.target);
+    event.target?.playVideo();
+    setIsPlaying(true);
+    setDuration(event.target?.getDuration());
+
+    event.target?.setVolume(volume);
+    if (isMuted) {
+      event.target?.mute();
+    } else {
+      event.target?.unMute();
+    }
+
+    clearInterval(timer);
+    setTimer(setInterval(() => {
+      setCurrentTime(event.target?.getCurrentTime() ?? 0);
+    }, 100));
+  }, [isMuted, volume, timer]);
+
+  const onEnd = useCallback(() => {
+    if (repeatMode === 'NO_REPEAT') {
+      playlist.next();
+    } else if (repeatMode === 'REPEAT_ONE') {
+      player?.playVideo();
+    } else {
+      playlist.next(true);
+    }
+  }, [player, playlist, repeatMode]);
+
+  const onTogglePlay = useCallback(() => {
+    if (isPlaying) {
+      player?.pauseVideo();
+    } else {
+      player?.playVideo();
+    }
+    setIsPlaying(!isPlaying);
+  }, [isPlaying, player]);
```

Comment:
```suggestion
  const onTogglePlay = useCallback(() => {
    setIsPlaying((prevPlaying) => {
      if(prevPlaying) player?.pauseVideo();
      else player?.playVideo();

      return !prevPlaying;
    });
  }, [player]);
```
`isPlaying` dependency 삭제

Reply:
toggle 도 마찬가지로 부탁이염

Reply:
@Su-Yong toggle 부분도 작업해주실 수 있나요?